### PR TITLE
Create NavItem component

### PR DESCRIPTION
**Pull request by fa-901**
## Description

Created NavItem component according to the requirements in issue [5911](https://github.com/nodejs/nodejs.org/issues/5911). Styled by matching the Figma styles to its equivalent Tailwind class.

## Validation

Added all styles in light / dark mode for both Nav & footer types. Some screenshots below

image
image

## Related Issues

Addresses issue [5911](https://github.com/nodejs/nodejs.org/issues/5911)

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/sections/NavItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { ArrowUpRightIcon } from '@heroicons/react/24/solid';
+import classNames from 'classnames';
+import Link from 'next/link';
+import { useMemo, type FC } from 'react';
+
+import styles from './index.module.css';
+
+export enum NavItemType {
```
Comment (ovflowd) [inline components/sections/NavItem/index.tsx]: Can this be a string type? `type NavItemType = 'nav' |'footer';`

File: components/sections/NavItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { ArrowUpRightIcon } from '@heroicons/react/24/solid';
+import classNames from 'classnames';
+import Link from 'next/link';
```
Comment (ovflowd) [inline components/sections/NavItem/index.tsx]: Please use LocalizedLink instead of next/link

File: components/sections/NavItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }, [href, type]);
+
+  return (
+    <Link href={href} className={classNames(styles.navItem, styles[type])}>
+      <span className={styles.label}>{label}</span>
+      {showIcon && <ArrowUpRightIcon className={styles.icon} />}
+    </Link>
+  );
+};
+export default NavItem;
```
Comment (ovflowd) [inline components/sections/NavItem/index.tsx:35]:
```suggestion

export default NavItem;
```

File: components/sections/NavItem/index.tsx (hunk 1 above)
Comment (fa-901) [reply components/sections/NavItem/index.tsx]: changed to string. But imho, I think using an enum would make it easier to refactor later (if needed)

**Review by rodion-arr — APPROVED** [review]

File: components/sections/NavItem/index.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import type { Meta as MetaObj, StoryObj } from '@storybook/react';
+
+import NavItem from './index';
+
+type Story = StoryObj<typeof NavItem>;
+type Meta = MetaObj<typeof NavItem>;
+
+export const Default: Story = {
+  args: {
+    href: '/about',
```
Comment (AugustinMauroy) [inline components/sections/NavItem/index.stories.tsx]:
```suggestion
    href: '/learn',
```

File: components/sections/NavItem/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    label: 'Learn',
+  },
+};
+
+export const WithExternalLink: Story = {
+  args: {
+    href: 'https://nodejs.org/en',
```
Comment (AugustinMauroy) [inline components/sections/NavItem/index.stories.tsx:17]:
```suggestion
    href: 'https://nodejs.org/en/learn',
```

File: components/sections/NavItem/index.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    label: 'Learn',
+  },
+};
+
+export const FooterItem: Story = {
+  args: {
+    href: '/about',
```
Comment (AugustinMauroy) [inline components/sections/NavItem/index.stories.tsx:24]:
```suggestion
    href: 'https://trademark-policy.openjsf.org/',
```

**Review by AugustinMauroy — APPROVED** [review]
LGTM ! just we can add real link for polish ✨

File: components/sections/NavItem/index.stories.tsx (hunk 5 above)
Comment (ovflowd) [reply components/sections/NavItem/index.stories.tsx:17]: Why these link suggestions? What's wrong with the current ones?

File: components/sections/NavItem/index.stories.tsx (hunk 6 above)
Comment (ovflowd) [reply components/sections/NavItem/index.stories.tsx:24]: This item is not supposed to be an external link

File: components/sections/NavItem/index.stories.tsx (hunk 5 above)
Comment (AugustinMauroy) [reply components/sections/NavItem/index.stories.tsx:17]: I propose that just to fit whit text

File: components/sections/NavItem/index.stories.tsx (hunk 6 above)
Comment (AugustinMauroy) [reply components/sections/NavItem/index.stories.tsx:24]: The text is "trademark policy" so I propose to put the  _"correct"_ link

Comment (ovflowd) [reply components/sections/NavItem/index.stories.tsx:24]: This is just a Storybook, Augustin. For all I know the link could point to google.com;

File: components/sections/NavItem/index.tsx (hunk 1 above)
Comment (ovflowd) [reply components/sections/NavItem/index.tsx]: Not really, TypeScript ensures that the values follow the suitable types. Also, enums are usually a discouraged practice. As in practical terms they're not a type, but an object.

File: components/sections/NavItem/index.module.css [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    &:active {
+      @apply bg-green-600;
+
+      .label {
+        @apply text-white;
+      }
+
+      .icon {
+        @apply text-white
+        opacity-50;
```
Comment (ovflowd) [inline components/sections/NavItem/index.module.css]: ^ missing right indentation

File: components/sections/NavItem/index.module.css [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      }
+    }
+  }
+
+  &.footer {
+    .label {
+      @apply text-neutral-800
+      dark:text-white;
```
Comment (ovflowd) [inline components/sections/NavItem/index.module.css]: ^ missing right indentation

File: components/sections/NavItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+type NavItemProps = {
+  href: string;
+  label?: string;
+  type?: NavItemType;
+};
+
+const NavItem: FC<NavItemProps> = ({ href, label, type = 'nav' }) => {
+  const showIcon = useMemo(() => {
+    return href.startsWith('http') && type === 'nav';
```
Comment (ovflowd) [inline components/sections/NavItem/index.tsx]:
```tsx
const showIcon = useMemo(() => 
  type === 'nav' && /^https?:\/\//.test(href),
  [href, type]
);
```

Can we simplify this function to what I've shown above? This regex is also what we use on the `LocalizedLink` component.

**Review by ovflowd — APPROVED** [review]
Re-approving. I just left a few final comments, and we're good to go! Thank you!

File: components/sections/NavItem/index.stories.tsx (hunk 4 above)
Comment (farhan-bs23) [reply components/sections/NavItem/index.stories.tsx]: updated

Comment (fa-901) [reply components/sections/NavItem/index.stories.tsx]: updated

File: components/sections/NavItem/index.module.css (hunk 7 above)
Comment (fa-901) [reply components/sections/NavItem/index.module.css]: updated

File: components/sections/NavItem/index.module.css (hunk 8 above)
Comment (fa-901) [reply components/sections/NavItem/index.module.css]: updated

File: components/sections/NavItem/index.tsx (hunk 9 above)
Comment (fa-901) [reply components/sections/NavItem/index.tsx]: updated

**Review by aymen94 — APPROVED** [review]

**Pull request merged**